### ISSUE TITLE
making the content of __all__ consistent with the API references in the docs. 

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4137]
One thing I'm a bit unsure about is that now in some cases as in the change of `_check_mesh` to `check_mesh`, how do we signal to the user and future developers that this function should be used internally (which also implies they can change without deprecation)? You can argue that it is not exposed in the API reference docs but some people don't look at that. One way could be to not include such functions in `__all__` and explicitly use the module.file name when calling the function as is now done with `check_mesh`. But then we'd need to make this consistent across modules (with the exceptions being _utils) making the content of `__all__` consistent with the API references in the docs. I'm not sure if this is the way to go so I'm happy to hear other thoughts

_Originally posted by @ymzayek in https://github.com/nilearn/nilearn/pull/4086#pullrequestreview-1719795351_

**Comment by bthirion** [comment]
Maybe it's a good motivation to maintain the leading underscore.

**Comment by Remi-Gau** [comment]
@ymzayek I tried to go through the inconsistencies between the declared public API of the different modules mentioned in their `__all__` and what we have in the user facing doc.

Did you have other inconsistencies in mind?

**Comment by ymzayek** [comment]
No I think you've captured them well!
Maybe hold off on the ones in the surface module for now until we see about integrating the new one.

**Comment by Remi-Gau** [comment]
yeah that was my thought on those as well

**Comment by Remi-Gau** [comment]
Extracting comments from #4157 

in `nilearn/reporting/__init__.py`

```python
# TODO consider making _get_clusters_table module public?
from nilearn.reporting._get_clusters_table import get_clusters_table
```

in `nilearn/image/__init__.py`

```python
# TODO move copy_img out of nilearn._utils..niimg
from .._utils.niimg import copy_img

# TODO move concat_niimgs out of nilearn._utils.niimg_conversions
from .._utils.niimg_conversions import concat_niimgs as concat_imgs
```

**Comment by Remi-Gau** [comment]
OK opening #4157 for review and will tackle the things I commented on in further PRs to keep PR / reviews smaller.

**Comment by Remi-Gau** [comment]
Can be closed once #4166 and #4167 are done
